h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1452,7 +1452,7 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
     GLMScore gs = makeScoringTask(adaptFrm,true,j);// doAll(names.length,Vec.T_NUM,adaptFrm);
     assert gs._dinfo._valid:"_valid flag should be set on data info when doing scoring";
     gs.doAll(names.length,Vec.T_NUM,gs._dinfo._adaptedFrame);
-    if (gs._computeMetrics)
+    if (computeMetrics && gs._computeMetrics)
```

Comment:
It seems like a better fix would be to propagate the flag to `makeScoringTask` - because GLMScore will actually run `_mb.perRow` to prepare the metadata for the metrics

Reply:
@michalkurka:  yes, wanted to minimize changes but this would be more robust. 
Will propagate then.